### PR TITLE
22 cumulative imputation links are incorrect for fic

### PR DESCRIPTION
**Pull request by AntonZogk**
# Summary

This pull request aims to fix some bugs as outlined in the #22  , tests and test data adapted accordingly 

# Checklists

This pull request meets the following requirements:

- [ ] installable with all dependencies recorded
- [x] runs without error
- [x] follows PEP8 and project specific conventions
- [x] appropriate use of comments, for example no descriptive comments
- [x] functions documented using Numpy style docstings
- [x] assumptions and decisions log considered and updated if appropriate
- [x] unit tests have been updated to cover essential functionality for a reasonable range of inputs and conditions
- [x] other forms of testing such as end-to-end and user-interface testing have been considered and updated as required
- [x] tests suite passes (locally as a minimum)
- [x] peer reviewed with review recorded

If you feel some of these conditions do not apply for this pull request, please
add a comment to explain why.

- installable with all dependencies recorded not applicable, the changes are for the low levels functions

### CODE REVIEW

File: src/apply_imputation_link.py [hunk 1]
```diff
@@ -55,34 +55,30 @@ def create_and_merge_imputation_values(
...
         "bir": {
             "intermediate_column": "bir",
-            "marker": "BIR",
+            "marker": "bir",
             "fill_column": target,
             "fill_method": "bfill",
             "link_column": cumulative_backward_link,
         },
         "fic": {
-            # FIC only works if the C is in the first period of the business being
```
Comment (rowanhemsi) [inline src/apply_imputation_link.py:79]: Has this issue been resolved?

File: src/cumulative_imputation_links.py [hunk 2]
```diff
@@ -46,7 +46,12 @@ def get_cumulative_links(
 
     dataframe["imputation_group"] = (
         (
-            (dataframe["missing_value"].diff(time_difference) != 0)
+            (
+                dataframe["imputation_marker"]
+                .ne(dataframe["imputation_marker"].shift().bfill())
+                .astype(int)
```
Comment (rowanhemsi) [inline src/cumulative_imputation_links.py:52]: Lines 47 to 57 are a bit hard to follow. Consider defining this condition first before applying it

File: tests/test_cumulative_imputation_links.py [hunk 3]
```diff
@@ -15,15 +15,16 @@ def cumulative_links_test_data():
 class TestComulativeLinks:
```
Comment (rowanhemsi) [inline tests/test_cumulative_imputation_links.py:15]: Typo in cumulative

File: tests/test_cumulative_imputation_links.py [hunk 4]
```diff
@@ -36,29 +37,38 @@ def test_get_cumulative_links_forward(self, cumulative_links_test_data):
...
         actual_output = get_cumulative_links(
             input_data,
-            "b",
+            "f",
             "strata",
             "reference",
             "target",
             "period",
-            "backward_imputation_link",
+            "forward_imputation_link",
```
Comment (rowanhemsi) [inline tests/test_cumulative_imputation_links.py:66]: It there a reason this has changed from backward to forward?

**Review by rowanhemsi — COMMENTED** [review]
Not very familiar with the bug so would want to check with someone else whether the test data covers it. Have left a couple of questions and suggestions on the code but most of them are very small

File: src/apply_imputation_link.py (hunk 1 above)
Comment (AntonZogk) [reply src/apply_imputation_link.py:79]: @robertswh i think it is resolved?

File: tests/test_cumulative_imputation_links.py (hunk 4 above)
Comment (AntonZogk) [reply tests/test_cumulative_imputation_links.py:66]: Good spot, this should be backward, will update it

File: src/cumulative_imputation_links.py (hunk 2 above)
Comment (AntonZogk) [reply src/cumulative_imputation_links.py]: You are absolutely right, i defined them in the top, also added a TODO line since similar conditions are applied 2-3 times in other functions

**Comment by AntonZogk** [comment]
> Not very familiar with the bug so would want to check with someone else whether the test data covers it. Have left a couple of questions and suggestions on the code but most of them are very small

We changed the approach and this is reflected to the test data ( added a new column for marker).  As described at #22 , instead of looking at time difference , the code now looks at imputation marker difference. So yes it is covered.

File: src/apply_imputation_link.py (hunk 1 above)
Comment (AntonZogk) [reply src/apply_imputation_link.py:79]: By accident i removed these comments from the code, i will re add them in the original code. 

This pull request is not related what is being mentioned in the comments, the comments are related to the manual construction methods and things we must be aware of.

**Review by KateyMatthews — APPROVED** [review]

**Comment by AntonZogk** [comment]
Rebased the branch to main to resolve conflicts caused by changing folder name src -> mbs_results
